internal/datastore/common/gc.go

```diff
@@ -196,7 +196,9 @@ func RunGarbageCollection(gc GarbageCollector, window, timeout time.Duration) er
 	}
 
 	defer func() {
-		err := gc.UnlockAfterGCRun(ctx)
+		// NOTE: this uses a separate context to run the DB unlock logic, ensuring that
+		// a timeout in the parent context won't prevent a lock release.
+		err := gc.UnlockAfterGCRun(context.Background())
```

Comment:
wouldn't this be clearer?
```
err := gc.UnlockAfterGCRun()
```
and then the implementations for the datastores can handle it as appropriate (i.e. it may make sense to have an internal timeout and kill the session)

Reply:
Not sure if it is clearer (as a caller I'd expect a context), but I can document it